### PR TITLE
Compressor

### PR DESCRIPTION
**Pull request by Tom32i**
The goal of this branch is to reduce the size of the data transferred from the client to the server.
Especially data that is being sent a lot: the position of the players.

## Optimisation strategies:
* Send float as integers
* Send integer indexed arrays instead of key/value objects

### CODE REVIEW

File: src/client/controller/GameController.js [hunk 1]
```diff
@@ -351,13 +355,28 @@ GameController.prototype.onBonusClear = function(e)
  *
  * @param {Event} e
  */
+GameController.prototype.onPosition = function(e)
+{
+    var avatar = this.game.avatars.getById(e.detail[0]),
+        position = this.compressor.decompressPosition(e.detail[1][0], e.detail[1][1]);
```
Comment (DaphDaOne) [inline src/client/controller/GameController.js]: You can compute position in the avatar condition, it won't be used if the condition is false anyway

File: src/client/controller/GameController.js [hunk 2]
```diff
@@ -351,13 +355,28 @@ GameController.prototype.onBonusClear = function(e)
...
+ */
 GameController.prototype.onPoint = function(e)
 {
-    var data = e.detail,
-        avatar = this.game.avatars.getById(data.avatar);
+    var avatar = this.game.avatars.getById(e.detail[0]),
+        position = this.compressor.decompressPosition(e.detail[1][0], e.detail[1][1]);
 
     if (avatar) {
-        avatar.addPoint(data.point);
```
Comment (DaphDaOne) [inline src/client/controller/GameController.js]: same here :)

File: src/client/repository/RoomRepository.js [hunk 3]
```diff
@@ -457,6 +457,7 @@ RoomRepository.prototype.onGameEnd = function(e)
  */
 RoomRepository.prototype.onVote = function(e)
 {
+    console.log(e);
```
Comment (DaphDaOne) [inline src/client/repository/RoomRepository.js]: debug spotted ! :)

File: src/client/controller/GameController.js (hunk 1 above)
Comment (Tom32i) [inline src/client/controller/GameController.js]: :bulb: :+1:

**Comment by Tom32i** [comment]
Thanks for the review!

**Pull request merged**